pkg/snmp/traps/config_test.go

```diff
@@ -6,68 +6,98 @@
 	assert.NoError(t, err)
 	assert.Equal(t, uint16(1234), config.Port)
-	assert.Equal(t, defaultStopTimeout, config.StopTimeout)
+	assert.Equal(t, 12, config.StopTimeout)
+	assert.Equal(t, []string{"public"}, config.CommunityStrings)
+	assert.Equal(t, "127.0.0.1", config.BindHost)
+	assert.Equal(t, []UserV3{
+		{
+			Username:     "user",
+			AuthKey:      "password",
+			AuthProtocol: "MD5",
+			PrivKey:      "password",
+			PrivProtocol: "AES",
+		},
+	}, config.Users)
 
-	params := config.BuildV2Params()
+	params, err := config.BuildSNMPParams()
+	assert.NoError(t, err)
 	assert.Equal(t, uint16(1234), params.Port)
-	assert.Equal(t, gosnmp.Version2c, params.Version)
+	assert.Equal(t, gosnmp.Version3, params.Version)
 	assert.Equal(t, "udp", params.Transport)
 	assert.NotNil(t, params.Logger)
+	assert.Equal(t, gosnmp.UserSecurityModel, params.SecurityModel)
+	assert.Equal(t, &gosnmp.UsmSecurityParameters{
+		UserName:                 "user",
+		AuthoritativeEngineID:    expectedEngineID,
+		AuthenticationProtocol:   gosnmp.MD5,
+		AuthenticationPassphrase: "password",
+		PrivacyProtocol:          gosnmp.AES,
+		PrivacyPassphrase:        "password",
+	}, params.SecurityParameters)
 }
 
-func TestDefaultPort(t *testing.T) {
-	Configure(t, Config{
-		CommunityStrings: []string{"public"},
-	})
-	config, err := ReadConfig()
-	assert.NoError(t, err)
-	assert.Equal(t, defaultPort, config.Port)
-}
-
-func TestCommunityStringsEmpty(t *testing.T) {
-	Configure(t, Config{
-		CommunityStrings: []string{},
-	})
-	_, err := ReadConfig()
-	assert.Error(t, err)
-}
-
-func TestCommunityStringsMissing(t *testing.T) {
+func TestMinimalConfig(t *testing.T) {
 	Configure(t, Config{})
-	_, err := ReadConfig()
-	assert.Error(t, err)
-}
-
-func TestDefaultStopTimeout(t *testing.T) {
-	Configure(t, Config{
-		CommunityStrings: []string{"public"},
-	})
-	config, err := ReadConfig()
+	config, err := ReadConfig("")
 	assert.NoError(t, err)
-
+	assert.Equal(t, uint16(162), config.Port)
 	assert.Equal(t, 5, config.StopTimeout)
+	assert.Equal(t, []string{}, config.CommunityStrings)
+	assert.Equal(t, "localhost", config.BindHost)
+	assert.Equal(t, []UserV3{}, config.Users)
+
+	params, err := config.BuildSNMPParams()
+	assert.NoError(t, err)
+	assert.Equal(t, uint16(162), params.Port)
+	assert.Equal(t, gosnmp.Version2c, params.Version)
+	assert.Equal(t, "udp", params.Transport)
+	assert.NotNil(t, params.Logger)
+	assert.Equal(t, nil, params.SecurityParameters)
 }
 
-func TestStopTimeout(t *testing.T) {
+func TestDefaultUsers(t *testing.T) {
 	Configure(t, Config{
 		CommunityStrings: []string{"public"},
 		StopTimeout:      11,
 	})
-	config, err := ReadConfig()
+	config, err := ReadConfig("")
 	assert.NoError(t, err)
 
 	assert.Equal(t, 11, config.StopTimeout)
 }
+
+func TestBuildAuthoritativeEngineID(t *testing.T) {
+	Configure(t, Config{})
+	config, err := ReadConfig(mockedHostname)
+	assert.NoError(t, err)
+	assert.Equal(t, expectedEngineID, config.authoritativeEngineID)
+}
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement.  -->
**🥜 nitpick:** ‏I think it can help to test with multiple hostnames and assert they lead to different engineID.

Reply:
Addressed